models/settings_helpers.go

```diff
@@ -292,6 +304,31 @@ func ValidateSettings(params *ChangeSettingsParams) error {
 		}
 	}
 
+	checkCases = []struct {
+		dur       time.Duration
+		fieldName string
+	}{
+		{params.STTCheckIntervals.RareInterval, "rare_interval"},
+		{params.STTCheckIntervals.StandardInterval, "standard_interval"},
+		{params.STTCheckIntervals.FrequentInterval, "frequent_interval"},
+	}
+	for _, v := range checkCases {
+		if v.dur == 0 {
+			continue
+		}
+
+		if _, err := validators.ValidateSTTCheckInterval(v.dur); err != nil {
+			switch err.(type) {
+			case validators.DurationNotAllowedError:
+				return fmt.Errorf("%s: should be a natural number of seconds", v.fieldName)
+			case validators.MinDurationError:
+				return fmt.Errorf("%s: minimal resolution is 1s", v.fieldName)
+			default:
+				return fmt.Errorf("%s: unknown error for", v.fieldName)
+			}
+		}
+	}
+
```

Comment:
nit: this code is same as the validation for `MetricsResolutions` and could be extracted out to another function. I would suggest that you do it now if you feel it can be done quickly otherwise just leave a TODO comment.

Reply:
You're right, but It would be a very specific code. I'll leave a TODO for now